lib/tasker.js

```diff
@@ -1,154 +1,81 @@
 /* global Zinnia */
 
-import { ActivityState } from './activity-state.js'
-import { encodeHex } from '../vendor/deno-deps.js'
-import { assertOkResponse, assertRedirectResponse } from './http-assertions.js'
-import { getRandomnessForSparkRound } from './drand-client.js'
+import { MERIDIAN_CONTRACT, SPARK_BASE_URL } from './constants.js'
+import { assertOkResponse } from './http-assertions.js'
 import { assertEquals, assertInstanceOf } from 'zinnia:assert'
 
 /** @typedef {{cid: string; minerId: string;}} RetrievalTask */
-/** @typedef {RetrievalTask & { key: string}} KeyedRetrievalTask */
 
 export class Tasker {
-  #lastRoundUrl
   /** @type {Task[]} */
-  #remainingRoundTasks
   #fetch
-  #activity
 
   /**
-   * @param {object} args
-   * @param {globalThis.fetch} args.fetch
-   * @param {ActivityState} args.activityState
-   */
+         * @param {object} args
+         * @param {globalThis.fetch} args.fetch
+        */
   constructor ({
-    fetch = globalThis.fetch,
-    activityState = new ActivityState()
+    fetch = globalThis.fetch
   } = {}) {
     this.#fetch = fetch
-    this.#activity = activityState
-
-    this.maxTasksPerRound = 360
-
-    // TODO: persist these two values across module restarts
-    // Without persistence, after the Spark module is restarted, it will start executing the same
-    // retrieval tasks we have already executed
-    this.#lastRoundUrl = 'unknown'
-    this.#remainingRoundTasks = []
   }
 
   /**
-   * @returns {Task | undefined}
-   */
-  async next () {
-    await this.#updateCurrentRound()
-    return this.#remainingRoundTasks.pop()
-  }
-
-  async #updateCurrentRound () {
-    console.log('Checking the current SPARK round...')
-    let res = await this.#fetch('https://api.filspark.com/rounds/current', {
+        * @param {object} args
+        * @param {number} [args.roundId]
+        * @param {string} [args.minerId]
+        * @param {number} [args.maxTasks]
+        * @returns {Promise<RetrievalTask[] | undefined>}
```

Comment:
```suggestion
        * @returns {Promise<RetrievalTask[]>}
```

I believe this promise can't resolve with `undefined`